packages/opencode/src/file/index.ts

```diff
@@ -87,7 +87,26 @@ export namespace File {
     const tops = ["image", "audio", "video", "font", "model", "multipart"]
     if (tops.includes(top)) return true
 
-    if (type === "application/octet-stream") return true
+    if (type === "application/octet-stream") {
+      const ext = file.name?.split(".").pop()?.toLowerCase()
+      const textExtensions = [
+        "py",
+        "go",
+        "rb",
+        "swift",
+        "kt",
+        "rs",
+        "scala",
+        "r",
+        "lua",
+        "pl",
+        "sh",
+        "bash",
+        "zsh",
+        "fish",
+      ]
```

Comment:
Where did this list come from? Is it comprehensive? Because I think I've seen this issue with other file types as well.

Reply:
I think the better change is to just be more explicit about only encoding image and zip files, nothing else. so removing the `application/octet-stream` check altogether, treat those as text.